service/application.go

```diff
@@ -254,6 +256,24 @@ func (app *Application) setupConfigurationComponents(ctx context.Context) error
 	}
 
 	app.service = service
+
+	// If provider is watchable start a goroutine watching for updates.
+	if watchable, ok := app.parserProvider.(parserprovider.Watchable); ok {
+		go func() {
+			err := watchable.WatchForUpdate()
```

Comment:
I guess we expected this method to block until an error occurs? If that is the case I think this should be explicit in the Watchable interface.